setup.py

```diff
@@ -45,6 +45,8 @@
     "protobuf>=3.20.2,<6.0.0dev,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5",
     "grpc-google-iam-v1 >= 0.12.4, < 1.0.0dev",
     "grpcio-status >= 1.33.2",
+    "opentelemetry-api",
+    "opentelemetry-sdk",
 ]
 extras = {"libcst": "libcst >= 0.3.10"}
```

Comment:
Adding opentelemetry as an optional dependency would require peppering the code base with 

```
try:
   from opentelemetry import X, Y, Z
   open_telemetry_imported = True
except ImportError:
   open_telemetry_imported = False

if open_telemetry_imported:
   // use open telemetry
```

And this gets messier with subclasses of opentelemetry classes like google/cloud/pubsub_v1/open_telemetry/context_propagation.py

where we have

```

from opentelemetry.propagators.textmap import Setter

class OpenTelemetryContextSetter(Setter):
   // class definition
```

And OpenTelemetryContextSetter is imported into other files in the repository. I would rather prefer just adding opentelemetry as a non-optional dependency. I think the addition of one dependency, which negates the need to pepper  try/except + if import clauses would be an okay tradeoff. Plmk if that works for you @hongalex @parthea

Reply:
Discussed this offline. We decided to stick with hard dependencies since the opentelemetry dependency is only 2.8M (excluding transitive dependencies). There isn't too much of a size increase in the client library. The advantages is we avoid convoluted try / except blocks and if import_success / else blocks in the code base.